### PR TITLE
Heroku added time offset + confirmation on dyno not using credit when sleeping

### PR DESCRIPTION
**Pull request by huizichin**
i used to host discord bot on heroku and it doesnt use credit when it is not active (called by discord api)
this also goes to this script it did not use any credit so it shouldnt be a worry, it would only use credit during the check in operation which only take a few seconds
image
here is a image after having it running for a days(since yesterday i created that pull request

### CODE REVIEW

**Comment by am-steph** [comment]
image
It notes here that apps that only utilize a free worker dyno don't sleep. I'm hosting a bot currently on Heroku and I can tell you that it does consume hours. 
image

File: main.py [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+tt = dt.datetime.strptime(run_time, '%H:%M')
+ha = dt.timedelta(hours = utc_timezone)
+lt = tt + ha
+fot = lt.strftime("%H:%M")
 
 def job():
     randomSleep = random.randint(10,300)
     print("Sleeping for: %ds" % randomSleep)
     time.sleep(randomSleep)
     exec(open('./run.py').read())
```
Comment (am-steph) [inline main.py:17]: run.py has random sleep, if you are adding random sleep in main.py, just directly execute genshin-os.py

**Review by am-steph — CHANGES_REQUESTED** [review]
change run.py to genshin-os.py

File: main.py [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+    rs = True
 
-schedule.every().day.at("06:00").do(job)
+schedule.every().day.at(fot).do(job)
 
 print("Script Started")
```
Comment (am-steph) [inline main.py:21]: Missing loop to run pending, it'll crash after printing this line

**Comment by huizichin** [comment]
weird enough tho it is not counting it into credit i have this running for 2 days
image

File: main.py (hunk 2 above)
Comment (huizichin) [reply main.py]: there is loop?

**Comment by huizichin** [comment]
nvm it did charge me just took them forever to update
image
here i updated it to use sleep only instead of that scheduler thingy uhh hopefully it work

**Comment by huizichin** [comment]
hmm it doesnt seem to work
it did send the webhook which is good it mean it triggered when i wanted it to but it showed this
image

**Comment by huizichin** [comment]
and it still charged me
image
though my discord bot seem to not been using much time tho weird

**Comment by am-steph** [comment]
You could continue trying to get Heroku dyno working, but I think its just easier to use Wayscript for those looking for a free method and its much easier to set up. Otherwise Heroku Scheduler is better to use to run on off scripts like this.

**Pull request closed without merging**